### PR TITLE
(FACT-2193) Fix trace.rb acceptance test

### PR DESCRIPTION
**Pull request by oanatmaria**

### CODE REVIEW

File: lib/custom_facts/core/logging.rb [hunk 1]
```diff
@@ -119,7 +119,10 @@ def format_exception(exception, message, trace)
           arr << message
         end
 
-        arr.concat(exception.backtrace) if trace
+        if trace
+          arr << 'backtrace:'
```
Comment (sebastian-miclea) [inline lib/custom_facts/core/logging.rb]: Could you add a test for this? It would be nice to catch this earlier and not when running acceptance tests.

Comment (oanatmaria) [reply lib/custom_facts/core/logging.rb]: Done

File: lib/util/exceptions.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+module Facter
+  module Util
+    class ResolveCustomFactError < StandardError; end
```
Comment (gimmyxd) [inline lib/util/exceptions.rb]: Error classes should not be Utils, you can either declare them in the Facter module or have them in an `erros/exceptions` folders

File: lib/custom_facts/core/resolvable.rb [hunk 3]
```diff
@@ -76,7 +76,8 @@ def value
         nil
       rescue StandardError => e
         LegacyFacter.log_exception(e, "Error while resolving custom fact #{qualified_name}: #{e.message}")
-        exit 1
+        at_exit { exit 1 }
```
Comment (gimmyxd) [inline lib/custom_facts/core/resolvable.rb]: do you need to `exit 1` if you're raising error? Is this error rescued somewhere else?

File: lib/custom_facts/util/fact.rb [hunk 4]
```diff
@@ -163,7 +163,11 @@ def sort_by_weight(resolutions)
 
       def find_first_real_value(resolutions)
         resolutions.each do |resolve|
-          value = resolve.value
+          begin
+            value = resolve.value
+          rescue Facter::Util::ResolveCustomFactError => _e
```
Comment (gimmyxd) [inline lib/custom_facts/util/fact.rb]: nit: you don't need `_e` if you're not going to use the error

Comment (oanatmaria) [reply lib/custom_facts/util/fact.rb]: done

File: lib/util/exceptions.rb (hunk 2 above)
Comment (oanatmaria) [reply lib/util/exceptions.rb]: Done

**Review by BogdanIrimie — APPROVED** [review]

**Pull request merged**